language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -216,6 +222,12 @@ genDefDataType curModName tpls def = case unTypeConName (dataTypeCon def) of
             ] ++
             map ("  " <>) (onHead ("export const decoder = " <>) serDesc) ++
             ["}"]
+        genBranch ((VariantConName cons, t), i) =
+          let (typ, ser) = (genType curModName t) in
+          ( (if i == 0 then "     " else "  |  ") <> "{ tag: '" <> cons <> "'; value: " <> typ <> " }"
```

Comment:
We can put the `|` even before the first case, like in OCaml. Let's do that and avoid the special case.

Reply:
Ooh! Testing. If all good will fix.